### PR TITLE
Initial draft for systemd process

### LINKED ISSUE #1772: OS boot/process management: systemd
Given the importance of process management for CockroachDB deployments, we need to:

- [ ] 1. Create and dogfood a sample `systemd` configuration suitable for production.
- [ ] 2. Expose the config in our docs.

For 2, we have two options:
- (Minimum) Keep manual deployment tutorials as-is but emphasize the importance of using a process manager and reference the sample config. 
- (Preferred) Adjust all of our non-orchestration tutorials to let the user toggle between instructions using `systemd` (new) vs. manual (current). We could do this for other tutorials as well, e.g., rolling upgrades.

Resources:
- [`systemd` wiki](https://www.freedesktop.org/wiki/Software/systemd/)
- Sample `systemd` configuration
    - [From forum](https://forum.cockroachlabs.com/t/ubuntu-service-script-to-start-a-cockroachdb-node-on-boot/1419)
    - [Internal](https://github.com/cockroachlabs/production/pull/483)

### PR DESCRIPTION
**Pull request by Amruta-Ranade**
Closes #1772 

@jseldess @mberhault This is an initial draft of the systemd process. For now, I have updated only the Digital Ocean deployment guide. Once the content is finalized, I will port the changes to the other deployment guides.

To-Do:
- Figure out where to place the sample configuration templates
- Edit the "Scale the cluster" sections
- Port content to other deployment guides

(Update: All To-Dos completed)

Future to-dos:
- Figure out how backup/restore and upgrades work with systemd

### CODE REVIEW

**Comment by bdarnell** [comment]
Review status: 0 of 6 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 6 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjxreYDAuTdvwds-W_:-LCjxreZ3OYTB34Zeh49:b-ynaqkx) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L6)):*
> ```
> 
> [Service]
> Type=simple
> ```

We support the systemd notification protocol, so this should be `Type=notify`. 

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 8 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjy9fu8VQZF6Gd8bYf:-LCjy9fu8VQZF6Gd8bYg:bef6i1h) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L8)):*
> ```
> Type=simple
> WorkingDirectory=/var/lib/cockroach
> ExecStart=/usr/local/bin/cockroach start --insecure --host=<node1 address> --join=<node1 address>:26257,<node2 address>:26257,<node3 address>:26257 --cache=25%% --max-sql-memory=25%%
> ```

Use `.25` instead of `25%%`.

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 9 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjy4-x7zC618mmcZiB:-LCjy4-x7zC618mmcZiC:b-d6ryde) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L9)):*
> ```
> WorkingDirectory=/var/lib/cockroach
> ExecStart=/usr/local/bin/cockroach start --insecure --host=<node1 address> --join=<node1 address>:26257,<node2 address>:26257,<node3 address>:26257 --cache=25%% --max-sql-memory=25%%
> ExecStop=/usr/local/bin/cockroach quit --insecure --host=<node1 address>
> ```

I'd prefer to have systemd stop the node via signal instead of `cockroach quit`. There are more things that can go wrong with the quit command. 

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-Z8M1PdzbFSKlK5K:-LCk-Z8N8RLixk_laFjs:bjsl66) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L15)):*
> ```
> RestartSec=10
> 
> # Allow user to stop node using systemctl stop cockroach (Prevent automatic restarts if node stopped manually)
> ```

What does "stopped manually" mean here? I think if systemd is used, we want it to restart automatically for any case except `systemctl stop`. 

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-NUb3g7KZ3KrV3El:-LCk-NUb3g7KZ3KrV3Em:bvz277k) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
> ```Markdown
> 7. Download the [sample configuration file]() for insecure clusters and save it in the `/etc/systemd/system/` directory.
> 
> 8. Customize the sample configuration template for your deployment:
> ```

Encouraging users to edit this file directly is unfortunate; does systemd provide an easy way to extend a service file without editing it?

---

*[_includes/prod_deployment/systemd-secure-start-nodes.md, line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjzJI14mDQKQAGIBuo:-LCjzJI14mDQKQAGIBup:b-dovzym) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-secure-start-nodes.md#L38)):*
> ```Markdown
>     ~~~
> 
> 6.  Move the `certs` directory to the `cockroach` directory.
> ```

Best practice would be that the `cockroach` user should not have access to the CA key, only the CA cert and the node key/cert. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by Amruta-Ranade** [comment]
Review status: 0 of 6 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 6 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjxreYDAuTdvwds-W_:-LD2elQkBUbI-wrFe1nn:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L6)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

We support the systemd notification protocol, so this should be `Type=notify`. 
</blockquote></details>

Done.

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 8 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjy9fu8VQZF6Gd8bYf:-LD2elxL1vudv-NtUABc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L8)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Use `.25` instead of `25%%`.
</blockquote></details>

Done.

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 9 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjy4-x7zC618mmcZiB:-LD2enFKDiXxxUdDd3VX:bgwfs2d) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L9)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

I'd prefer to have systemd stop the node via signal instead of `cockroach quit`. There are more things that can go wrong with the quit command. 
</blockquote></details>

Not sure what "signal" refers to. 

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-Z8M1PdzbFSKlK5K:-LD2evb40zOMRk-F7Tfr:bqeeaj2) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L15)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

What does "stopped manually" mean here? I think if systemd is used, we want it to restart automatically for any case except `systemctl stop`. 
</blockquote></details>

"Stopped manually" means the user explicitly stops the node. If `RestartPreventExitStatus=0 ` isn't specified, systemd will restart the node even if the user stopped the node intentionally. I can reword the statement as "Prevents automatic restarts if the user intentionally stops the node". Does it seem clearer?

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-NUb3g7KZ3KrV3El:-LD2ffQnBpDuOi2dy4je:bhecd3m) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Encouraging users to edit this file directly is unfortunate; does systemd provide an easy way to extend a service file without editing it?
</blockquote></details>

Not sure why we don't want users to edit this file directly. 

---

*[_includes/prod_deployment/systemd-secure-start-nodes.md, line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjzJI14mDQKQAGIBuo:-LD2fvBY-TblVrVjhvvD:b-7wxir3) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-secure-start-nodes.md#L38)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Best practice would be that the `cockroach` user should not have access to the CA key, only the CA cert and the node key/cert. 
</blockquote></details>

Again, not sure why they shouldn't have access since in the non-systemd process, they do have access to the entire certs directory. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by bdarnell** [comment]
Reviewed 2 of 6 files at r1, 2 of 2 files at r2.
Review status: 4 of 6 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 9 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjy4-x7zC618mmcZiB:-LD3WUSr5OyM-3nIsCLL:b-q6pggg) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L9)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Not sure what "signal" refers to. 
</blockquote></details>

"Signal" refers to killing the process with a unix signal, which is what systemd will do if no ExecStop command is  given. I think we should probably remove the ExecStop and rely on the default behavior, although we should also set `TimeoutStopSec` (to 60 seconds?) and test this. 

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-Z8M1PdzbFSKlK5K:-LD3Wlhk4Tx6_HpiSoZY:bnciyak) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L15)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

"Stopped manually" means the user explicitly stops the node. If `RestartPreventExitStatus=0 ` isn't specified, systemd will restart the node even if the user stopped the node intentionally. I can reword the statement as "Prevents automatic restarts if the user intentionally stops the node". Does it seem clearer?
</blockquote></details>

Stopped the node how? If they stop it with `systemctl stop cockroach`, systemd won't restart it. I don't think systemd users should be using non-systemd mechanisms to stop the process, and if they do, systemd should restart it. 

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-NUb3g7KZ3KrV3El:-LD3VJKf2x5pgzlboucy:bt0ns17) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Not sure why we don't want users to edit this file directly. 
</blockquote></details>

Because one day we'll need to publish a new version of this file, and then they'll need to figure out how to reconcile their changes with ours. If we had a separate config file, then they could just replace the service file completely. 

---

*[_includes/prod_deployment/systemd-secure-start-nodes.md, line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjzJI14mDQKQAGIBuo:-LD3VV2i0il15k1A9TEo:b-pzpsxw) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-secure-start-nodes.md#L38)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Again, not sure why they shouldn't have access since in the non-systemd process, they do have access to the entire certs directory. 
</blockquote></details>

The non-systemd processes are for test/development use that don't necessarily follow all the best practices. For example, we don't document creating a `cockroach` unix user in those cases. Using `systemd` is a more production-ready deployment so this is the place to follow all the best practices. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by jseldess** [comment]
*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-NUb3g7KZ3KrV3El:-LD8EbB81MLYwQ1Z1E5j:ba2ddxn) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Because one day we'll need to publish a new version of this file, and then they'll need to figure out how to reconcile their changes with ours. If we had a separate config file, then they could just replace the service file completely. 
</blockquote></details>

@bdarnell, it looks like we just asking the user to fill in the host and join flags. Are you saying there's a way to "extend" the file in these ways without actually editing it?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by bdarnell** [comment]
Review status: 4 of 6 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-NUb3g7KZ3KrV3El:-LD8FIi9Cm0Dp90z1KIf:b74vvcx) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
> Are you saying there's a way to "extend" the file in these ways without actually editing it?

Or if there's not we should build one. There's an `EnvironmentFile` option in systemd that might be suitable as a simple kind of configuration file. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by jseldess** [comment]
*[_includes/prod_deployment/systemd-secure-start-nodes.md, line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjzJI14mDQKQAGIBuo:-LD8G8G7DiaKpKPl1w8X:b-v4dv3v) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-secure-start-nodes.md#L38)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

The non-systemd processes are for test/development use that don't necessarily follow all the best practices. For example, we don't document creating a `cockroach` unix user in those cases. Using `systemd` is a more production-ready deployment so this is the place to follow all the best practices. 
</blockquote></details>

@bdarnell, in an earlier step, the user uploaded just the ca.crt, node.crt, and node.key files to the node. The ca.key file stays on the local machine. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

File: _includes/prod_deployment/systemd-secure-start-nodes.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    ~~~
+
+7.  Change the ownership of `Cockroach` directory to the user `cockroach`:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    chown -R cockroach.cockroach /var/lib/cockroach
+    ~~~
+
+8. Download the [sample configuration template]() for secure clusters and save it in the `/etc/systemd/system/` directory.
```
Comment (Amruta-Ranade) [inline _includes/prod_deployment/systemd-secure-start-nodes.md]: Include the file text here (refer to python app example).

**Comment by Amruta-Ranade** [comment]
*[_includes/prod_deployment/insecurecockroachdb.service, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-Z8M1PdzbFSKlK5K:-LD8Kv0B7xy12iiRs-MK:bbtinac) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L15)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Stopped the node how? If they stop it with `systemctl stop cockroach`, systemd won't restart it. I don't think systemd users should be using non-systemd mechanisms to stop the process, and if they do, systemd should restart it. 
</blockquote></details>

Remove the option, document the systemctl command (systemctl stop cockroach)

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by Amruta-Ranade** [comment]
*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-NUb3g7KZ3KrV3El:-LD8LTdd2BdDifILp7n2:bqiw1oa) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

> Are you saying there's a way to "extend" the file in these ways without actually editing it?

Or if there's not we should build one. There's an `EnvironmentFile` option in systemd that might be suitable as a simple kind of configuration file. 
</blockquote></details>

As discussed with Ben, okay to leave it as it is for now.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by Amruta-Ranade** [comment]
*[_includes/prod_deployment/systemd-secure-start-nodes.md, line 52 at r2](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCji0pkj2wwEMQfYJ6H-r2-52:-LDDo_Aq5e8r4Gy1XudK:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/4b1a60d6914f3ca5e209d44bd9ea6b84b29c4847/_includes/prod_deployment/systemd-secure-start-nodes.md#L52)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Include the file text here (refer to python app example).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by Amruta-Ranade** [comment]
*[_includes/prod_deployment/insecurecockroachdb.service, line 15 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCk-Z8M1PdzbFSKlK5K:-LDDmjjt5TSa2qKjKiTk:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L15)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Remove the option, document the systemctl command (systemctl stop cockroach)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by Amruta-Ranade** [comment]
*[_includes/prod_deployment/insecurecockroachdb.service, line 9 at r1](https://reviewable.io/reviews/cockroachdb/docs/3163#-LCjy4-x7zC618mmcZiB:-LDDoWXIDBkHU_KBg35b:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f0bf9631153378dacf2cb65d3db8eedb2960b397/_includes/prod_deployment/insecurecockroachdb.service#L9)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

"Signal" refers to killing the process with a unix signal, which is what systemd will do if no ExecStop command is  given. I think we should probably remove the ExecStop and rely on the default behavior, although we should also set `TimeoutStopSec` (to 60 seconds?) and test this. 
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

**Comment by jseldess** [comment]
Just ran through the insecure steps, and it all works great! I'll test secure as well, but first want to provide some feedback. Very sorry it's taken so long.

One suggestion not mentioned below is to make it clear that `systemd` will handle restarts in the case of nodes stopping. That's a main reason to use a system like this, but it's best to be explicit about the difference between manual and system. 

Of course, I test, and systemd nicely restarts a node that dies:



---

Review status: 0 of 6 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 6 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUwZXy2ks6CWdMgXUw:-LEUwZXy2ks6CWdMgXUx:b-jncphk) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/insecurecockroachdb.service#L6)):*
> ```
> [Service]
> Type=notify
> WorkingDirectory=/var/lib/cockroach
> ```

Is there a reason we need to put the node store here rather than in home? Just curious. Had to search a bit before I could find the logs. 

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 5 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUsvnW5_eL9a9yoeQm:-LEUsvnW5_eL9a9yoeQn:brrzhap) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L5)):*
> ```Markdown
> {{site.data.alerts.callout_info}}After completing these steps, nodes will not yet be live. They will complete the startup process and join together to form a cluster as soon as the cluster is initialized in the next step.{{site.data.alerts.end}}
> 
> 1. SSH to the machine where you want the node to run. Ensure you are logged in as the root user.
> ```

the root user > the `root` user

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 42 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUtoqS7SkCfjlmbAoG:-LEUtoqS7SkCfjlmbAoH:bhjt4le) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L42)):*
> ```Markdown
>     {% include copy-clipboard.html %}
>     ~~~ shell
>     chown cockroach /var/lib/cockroach
> ```

Add `$` prompt.

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 45 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUufhfDH4VqU2z_7nB:-LEUufhfDH4VqU2z_7nC:b-463ho0) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L45)):*
> ```Markdown
>     ~~~
> 
> 7. Download the [sample configuration template](/_includes/prod_deployment/insecurecockroachdb.service), or create the file yourself and copy the script into it:
> ```

I think it may be easier to give users a wget command to download the config directly to the server. Maybe something like this:

```
7.  Download the [sample configuration template](/_includes/prod_deployment/insecurecockroachdb.service):

    {% include copy-clipboard.html %}
    ~~~ shell
    $ wget -qO- https://xxxxx (where this is the url to the raw version of the file in our github repo)
    ~~~

    You can also create the file yourself and copy the script into it:
```


---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUukgn3Z9FX9pQO9lF:-LEUukgn3Z9FX9pQO9lG:b3663xe) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
> ```Markdown
> 7. Download the [sample configuration template](/_includes/prod_deployment/insecurecockroachdb.service), or create the file yourself and copy the script into it:
> 
>     ~~~ shell
> ```

Please add copy-to-clipboard.

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 66 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUvIFN-6ePzypiQJ3H:-LEUvIFN-6ePzypiQJ3I:bhjt4le) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L66)):*
> ```Markdown
>     {% include copy-clipboard.html %}
>     ~~~ shell
>     systemctl start insecurecockroachdb
> ```

Add `$` prompt.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163#-:-LEUy1weAGZNh9EjMiLV:b8rgyan)*

**Comment by Amruta-Ranade** [comment]
Review status: :shipit: complete! 0 of 0 LGTMs obtained

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 5 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUsvnW5_eL9a9yoeQm:-LEjeZBI5OxDBhQMk_XR:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L5)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

the root user > the `root` user
</blockquote></details>

Done.

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 42 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUtoqS7SkCfjlmbAoG:-LEjeZf6BkeVFoSkATPF:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L42)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add `$` prompt.
</blockquote></details>

Done.

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 45 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUufhfDH4VqU2z_7nB:-LEje_P3AITEEbfbmwr2:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L45)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think it may be easier to give users a wget command to download the config directly to the server. Maybe something like this:

```
7.  Download the [sample configuration template](/_includes/prod_deployment/insecurecockroachdb.service):

    {% include copy-clipboard.html %}
    ~~~ shell
    $ wget -qO- https://xxxxx (where this is the url to the raw version of the file in our github repo)
    ~~~

    You can also create the file yourself and copy the script into it:

</blockquote></details>

Done.

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 47 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUukgn3Z9FX9pQO9lF:-LEje_nT3lpVz9BySjGJ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L47)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Please add copy-to-clipboard.
</blockquote></details>

Done.

---

*[_includes/prod_deployment/systemd-insecure-start-nodes.md, line 66 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUvIFN-6ePzypiQJ3H:-LEjeaex8N5_ELdx9aTS:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/systemd-insecure-start-nodes.md#L66)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add `$` prompt.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*
<!-- Sent from Reviewable.io -->

**Comment by mberhault** [comment]
Review status: :shipit: complete! 0 of 0 LGTMs obtained

---

*[_includes/prod_deployment/insecurecockroachdb.service, line 6 at r3](https://reviewable.io/reviews/cockroachdb/docs/3163#-LEUwZXy2ks6CWdMgXUw:-LEk2tlyDpwL1wclxXwH:b-ljbbkr) ([raw file](https://github.com/cockroachdb/docs/blob/25391d34fa9a2c8edebf4d9c3d60a8392abb2972/_includes/prod_deployment/insecurecockroachdb.service#L6)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Is there a reason we need to put the node store here rather than in home? Just curious. Had to search a bit before I could find the logs. 
</blockquote></details>

`/var/lib/` is about the most standard place for service-based storage. eg: postgres ends up in `/var/lib/postgresql/`

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3163)*

File: _includes/prod_deployment/insecure-start-nodes.md [hunk 2]
```diff
@@ -47,4 +57,92 @@ For each initial node of your cluster, complete the following steps:
...
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ chown cockroach /var/lib/cockroach
+    ~~~
+
+7. Download the [sample configuration template](/_includes/prod_deployment/insecurecockroachdb.service):
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ wget -qO- https://xxxxx (where this is the url to the raw version of the file in our github repo)
```
Comment (jseldess) [inline _includes/prod_deployment/insecure-start-nodes.md]: https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/prod_deployment/insecurecockroachdb.service

File: _includes/prod_deployment/insecurecockroachdb.service [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[Unit]
```
Comment (jseldess) [inline _includes/prod_deployment/insecurecockroachdb.service]: You can remove this file from the PR now.

File: _includes/prod_deployment/secure-start-nodes.md [hunk 4]
```diff
@@ -48,4 +58,99 @@ For each initial node of your cluster, complete the following steps:
...
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ chown -R cockroach.cockroach /var/lib/cockroach
+    ~~~
+
+8.  Download the [sample configuration template](/_includes/prod_deployment/securecockroachdb.service):
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ wget -qO- https://xxxxx (where this is the url to the raw version of the file in our github repo)
```
Comment (jseldess) [inline _includes/prod_deployment/secure-start-nodes.md]: https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/prod_deployment/securecockroachdb.service

File: _includes/prod_deployment/securecockroachdb.service [hunk 5]
```diff
@@ -0,0 +1,16 @@
+[Unit]
```
Comment (jseldess) [inline _includes/prod_deployment/securecockroachdb.service]: You can remove this file from this PR now.

**Comment by Amruta-Ranade** [comment]
Update (06/14):
- The steps written for DO don't work as they are for GCE and throw some errors. Met with Marc to figure out why that is and need to try out myself to understand them better. Working on that now.

**Pull request merged**